stress/config.go

```diff
@@ -0,0 +1,145 @@
+package runner
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/BurntSushi/toml"
+	"github.com/influxdb/influxdb/client"
+)
+
+// tag is a struct that contains data
+// about a tag for in a series
+type tag struct {
+	Key   string `toml:"key"`
+	Value string `toml:"value"`
+}
+
+// tag is a struct that contains data
+// about a field for in a series
+type field struct {
+	Key  string `toml:"key"`
+	Type string `toml:"type"`
+}
+
+// series is a struct that contains data
+// about the series that will be written
+// during a stress test
+type series struct {
+	PointCount  int     `toml:"point_count"`
+	Measurement string  `toml:"measurement"`
+	SeriesCount int     `toml:"series_count"`
+	Tags        []tag   `toml:"tag"`
+	Fields      []field `toml:"field"`
+}
+
+// write is a struct that contains the business
+// logic for the stress test. e.g. where the
+// influxdb instance is running, what database
+// should points be written into
+type write struct {
+	Concurrency   int    `toml:"concurrency"`
+	BatchSize     int    `toml:"batch_size"`
+	BatchInterval string `toml:"batch_interval"`
+	Database      string `toml:"database"`
+	ResetDatabase bool   `toml:"reset_database"`
+	StartingPoint string `toml:"starting_time"`
+	Address       string `toml:"address"`
+	Precision     string `toml:"precision"`
+}
+
+// query is a struct that contains the logic for
+// a query that will be ran on during the stress
+// test
+type query struct {
+	Concurrency int    `toml:"concurrency"`
+	Measurement string `toml:"measurement"`
+	TagKey      string `toml:"tag_key"`
+	TimeFrame   string `toml:"time_frame"`
+	Statement   string `toml:"statement"`
+}
+
+// Config is a struct that is passed into the `Run()` function.
+type Config struct {
+	Write   write    `toml:"write"`
+	Series  []series `toml:"series"`
+	Queries []query  `toml:"query"`
+}
+
+// DecodeFile takes a file path for a toml config file
+// and returns a pointer to a Config Struct.
+func DecodeFile(s string) (*Config, error) {
+	t := &Config{}
+
+	if _, err := toml.DecodeFile(s, t); err != nil {
+		return nil, err
+	}
+
+	return t, nil
+}
+
+// seriesIter is a struct that contains a
+// series and a count, where count is the
+// number of points that have been written
+// for the series `s`
+type seriesIter struct {
+	s     *series
+	count int
+}
+
+// Iter returns a pointer to a seriesIter
+func (s *series) Iter() *seriesIter {
+	return &seriesIter{s: s, count: -1}
```

Comment:
The looping condition in [inner for loop of Run](https://github.com/influxdb/influxdb/blob/md-stress-config/stress/runner.go#L179-L180) relies on the bool returned by [`Next`](https://github.com/influxdb/influxdb/blob/md-stress-config/stress/config.go#L136-L145).